compute_endpoint/globus_compute_endpoint/endpoint/utils/__init__.py

```diff
@@ -155,3 +156,28 @@ def send_endpoint_startup_failure_to_amqp(amqp_creds: dict, msg: str | None = No
                 body=payload,
                 mandatory=True,
             )
+
+
+def user_input_select(prompt: str, options: list[str]) -> str | None:
+    """
+    Prompt the user to select from a list of options, and returns the item or
+    None if the user declines to select one
+    """
+    assert options, "Options list must not be empty"
+    if not (sys.stdin and not sys.stdin.closed) or not sys.stdin.isatty():
+        raise ValueError("Not interactive, cannot prompt for user input")
```

Comment:
Ah, so it _is_ `isatty()` protected.  "Okay."

Another option might be to emit to stderr if not a tty():

```python
if not (sys.stdin and not sys.stdin.closed):
    raise EOFError("No input (stdin closed); cannot collect user input")

if not sys.stdin.isatty():
    print("Reading batch data from stdin ...", file=sys.stderr)
```